### ISSUE TITLE
iPad OS horizontal mode cuts lattest message

### ISSUE THREAD

**Issue by alexander-potemkin** [mattermost/mattermost#17638]
#### Summary
When the app goes back from the background, it is like hiding the latest message - like in screenshot provide, 'System' actually had another message, saying that that another user joined in, but you can't see it, unless you manually pull this part of the screen up and hold it to read.

#### Steps to reproduce
Bring the app from the background. It works like once in 5 times, I guess.

#### Expected behavior
Keep showing all of the lines.

#### Observed behavior (that appears unintentional) 
Line is hidden.
<img width="1366" alt="MM line bug" src="https://user-images.githubusercontent.com/2323840/118664075-d7313c80-b7f9-11eb-8fe9-de6e008a0f0a.png">

#### Possible fixes
I can only guess, but could it be because of the extra line that the hardware keyboard creates?

**Comment by josephbaylon** [comment]
@alexander-akhmetov @amyblais This is has been fixed in v1.43. I'm not able to reproduce the issue anymore.

**Comment by amyblais** [comment]
@alexander-potemkin Please try v1.43 ^

**Comment by alexander-potemkin** [comment]
@amyblais , negative, same issue (just updated)

**Comment by alexander-potemkin** [comment]
@amyblais , negative - brought the app from background and got the last line hidden, just like reported. Latest app version.
Could you please, look into it?

**Comment by amyblais** [comment]
Does it reproduce consistently and what OS version are you on?

**Comment by amyblais** [comment]
@alexander-potemkin Have you still experienced this issue? Does it reproduce consistently and what OS version are you on?

**Comment by alexander-potemkin** [comment]
That one seems to be resolved, thank you!
Let me re-open it if I will notice any similar behaviour.

**Comment by alexander-potemkin** [comment]
![IMG_0240](https://user-images.githubusercontent.com/2323840/121406227-9050e580-c966-11eb-9710-30793130a2ea.PNG)
![IMG_0241](https://user-images.githubusercontent.com/2323840/121406235-934bd600-c966-11eb-91be-0b6cd6e577a2.PNG)

Got it again, here is how to reproduce:
- latest MM, iOS (14.4.2)
- iPad Pro 11, vertical mode
- in group message get the message you want to respond to (first screenshot)
- tap into the space to enter the text
- enjoy (second screenshot)!

**Comment by alexander-potemkin** [comment]
As a guess, the app calculates how much is shall move the text up, without considering the words hints place.
@amyblais

**Comment by amyblais** [comment]
@enahum Wondering if you have thoughts on this. This was fixed in v1.43 but the reporter is still able to reproduce this on v1.43.

**Comment by enahum** [comment]
Does this happen if the predictive keybord is turned off? It may just be that

**Comment by alexander-potemkin** [comment]
I will re-check on iPad 11, but at the meantime here is the same kind of issue on iPad 12.9, all latest versions, no keyboard and the message or smiles part get cut.

Does anybody from the developers dogfood, i.e. - use MatterMost on iPads at all? =)

<img width="1118" alt="Screenshot 2021-06-11 at 16 27 36" src="https://user-images.githubusercontent.com/2323840/121693790-1e95aa80-cad2-11eb-8024-ad81a8746e2d.png">

**Comment by alexander-potemkin** [comment]
Can’t reproduce it now, thanks!

**Comment by alexander-potemkin** [comment]
I’m sorry to say that it’s here again… =(
App: 1.44.1 (360), server: 5.36.1

**Comment by amyblais** [comment]
@enahum @josephbaylon Do we know why this has come back and how to prevent it from coming back in the future?

**Comment by enahum** [comment]
I am unable to reproduce, this time around it works fine for me.

This was caused by the scrollview being bound to follow the keyboard and in certain situations we need to re-bind that scrollview (very very very fragile). In 1.44 we were re-bind it scrollview qt all times which caused the issue where once you entered the thread view then the keyboard will cover messages on the channel. Now we are binding the scrollview for the channel only when it gets displayed.

I've tried switching channels with only text, images, reactions, rotating the device and have been unable to reproduce.. maybe @josephbaylon can

**Comment by alexander-potemkin** [comment]
![IMG_1084](https://user-images.githubusercontent.com/2323840/124601176-33b2ee80-de70-11eb-891e-0bd589c9378c.PNG)
![IMG_1083](https://user-images.githubusercontent.com/2323840/124601185-37467580-de70-11eb-983e-b0919458082e.PNG)
![IMG_1081](https://user-images.githubusercontent.com/2323840/124601188-3877a280-de70-11eb-89b6-4816beef729b.PNG)

And here comes the screenshots (got it re-appeared again)

**Comment by enahum** [comment]
@alexander-potemkin screenshots indeed show that is happening, so maybe your reproduction steps can help, cause is not happening for me

**Comment by alexander-potemkin** [comment]
@enahum , yes, I understand you. Unfortunately, it's not something I can reproduce reliably - I just use the app on daily basis on my iPad Pro 12.9 with Apple's Keyboard and sometimes without.

**Comment by alexander-potemkin** [comment]
It seems to start happening more and more often now; even my attempt to scroll things up doesn’t work - it returns back, cutting the last message - to read it I have to scroll things up with my finger and keep it there to read

**Comment by amyblais** [comment]
Opened a new ticket: https://mattermost.atlassian.net/browse/MM-36921

**Comment by alexander-potemkin** [comment]
Thanks. Any ETA? It's quite annoying..

**Comment by alexander-potemkin** [comment]
@enahum , it seems to happen more often, after I take iPad out of the keyboard, change it orientation vertical (to work around the bug with missing smiles), and then place it back horizontal to the keyboard stand.

**Comment by alexander-potemkin** [comment]
@amyblais , sorry to bother you, but the bugs with smiles and messages seems to be reported like 2 months ago and still with no bug fix on that. Do you have any time estimate or an idea when it could be worked out?

**Comment by alexander-potemkin** [comment]
@enahum , @amyblais , @anyone?

**Comment by amyblais** [comment]
I don't have a specific update, the ticket is in the Mobile team's queue. Maybe @enahum has more.

**Comment by alexander-potemkin** [comment]
Thank you, @amyblais ! @enahum ?
Is there anyone checking on the queue? I understanding, there are always a lot of things on the roadmap, but I’m quite depressed to see how the product is moving, in comparison to Slack - nor that I like cloud products, but Slack seems to be so much more polished (using both Mattermost & Slack, and compare it every day).

**Comment by enahum** [comment]
@alexander-potemkin I'm afraid the issue is not reproducing for me.. I've been using the app on an iPad Pro 11 for a few days now with and without an external keyboard.

Regardless of this, there are a few changes already in the next version (the one I'm using) so perhaps you could give the beta app a try and see if it is fixed for you with that one

**Comment by alexander-potemkin** [comment]
@enahum , thank you, but the issue happens on iPad Pro 12.9, not on iPad Pro 11 is there any chance you can test on it?
On iPad Pro 11 - do you have predictive & shortcuts enabled in the keyboard settings?

**Comment by enahum** [comment]
No predictive enabled, will give it a try.

For the iPad 12 it will be on a Simulator that is not very accurate in my opinion, but I can give it a quick pass

**Comment by alexander-potemkin** [comment]
If you have both predictive & shortcuts enabled, you get this little bar at the very bottom - if you don’t have it, it might be the thing that screw things.

iPad 12.9 on Simulator (with physical keyboard?) - I don’t believe it could work, but if there are no other options…

**Comment by enahum** [comment]
I'll have to try with a Sim for 12.9"

With shortcuts and predictive enabled works fine on my iPad Pro 11 with soft and hard keyboards

**Comment by alexander-potemkin** [comment]
Thank you. I only have issue with hardware keyboard and physical iPad Pro 12.9 - I hope you will be able to localize that issue.

**Comment by alexander-potemkin** [comment]
@enahum - I tried to work without this keyboard bar at the bottom, but Mattermost still ‘hides’ latests messages - so it’s not a matter of the keyboard bar; as a fool guess - probably the algorithms to calculate a proper position doesn’t work right on the bigger screen?

**Comment by alexander-potemkin** [comment]
one more observation - it happens after mattermost comes from the foreground and only on new messages - not sure if it’s a cold start or just return from the background; doesn’t seem to happen when switching between the channels (at least, in my last days of observation; will double check that particular statement).

**Comment by alexander-potemkin** [comment]
@enahum , it keeps happening on a daily basis and it’s quite frustrating; it is in place even after I send my own message - I still need to scroll to the top to see it, and when I release it - it’s hidden from me.

This issue was reported like 4 months ago and still nothing on it =(

**Comment by alexander-potemkin** [comment]
@amyblais , @enahum , any updates on that? it’s a really annoying thing on iPad and it’s like 5 months old now…

**Comment by lieut-data** [comment]
Thanks for reporting this issue and for the community discussion around it. After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work. If this is still relevant, please feel free to open a new issue with updated context and we'll be happy to take another look.
